common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt

```diff
@@ -54,19 +55,36 @@ fun JSONObject.createPackage(
     offeringIdentifier: String
 ): Package? {
     val packageIdentifier = getString("identifier")
-    // TODO handle INAPP products
     val productIdentifier = getString("platform_product_identifier")
-    val planIdentifier = getString("platform_product_plan_identifier")
-    val storeProducts: List<StoreProduct>? = productsById[productIdentifier]
-    val matchingProduct = storeProducts?.firstOrNull { storeProduct ->
-        storeProduct.purchaseOptions.firstOrNull { it.isBasePlan }?.id == planIdentifier
-    }
+    val planIdentifier = optString("platform_product_plan_identifier").takeIf { it.isNotEmpty() }
+
+    val matchingProduct = getMatchingProduct(productsById, productIdentifier, planIdentifier)
+
     return matchingProduct?.let { product ->
         val packageType = packageIdentifier.toPackageType()
         return Package(packageIdentifier, packageType, product, offeringIdentifier)
     }
 }
 
+private fun getMatchingProduct(
+    productsById: Map<String, List<StoreProduct>>,
+    productIdentifier: String,
+    planIdentifier: String?
+): StoreProduct? {
+    if (planIdentifier == null) {
+        // It could be an INAPP or a mis-configured subscription
+        // Try to find INAPP, otherwise null
+        return productsById[productIdentifier]
+            .takeIf { it?.size == 1 }
+            ?.takeIf { it[0].type == ProductType.INAPP }
+            ?.get(0)
```

Comment:
should we error if no INAPP product is found here? so people know they're misconfigured?

Reply:
We don't error for misconfigured subscriptions so I don't think we should do that. We actually don't error right now at all in BC4 in these cases and I don't think we should make that change for several reasons:
- We would also have a debug log if we Google doesn't return that OTP, so that debug log is already there.
- I am not sure if you mean an error log or returning an error in this function, but returning an error would mean that if someone is playing around with offerings and adding fake products they will make their apps error.
- Returning an error would mean that if someone updates to this version and has misconfigured products their app would stop loading offerings. Right now the apps try their best to load whatever is configured and add a debug log if a product ID doesn't match Google's configuration.

Reply:
yeah, definitely don't think we should return an error, just a log. and if that's covered already, i'm fine w it!

Reply:
this would be a good candidate for a "super-debuggy" log setting in the future